internal/eventstore/eventstore.go

```diff
@@ -208,12 +221,26 @@ func (es *Eventstore) LatestSequence(ctx context.Context, queryFactory *SearchQu
 }
 
 // InstanceIDs returns the instance ids found by the search query
-func (es *Eventstore) InstanceIDs(ctx context.Context, queryFactory *SearchQueryBuilder) ([]string, error) {
+func (es *Eventstore) InstanceIDs(ctx context.Context, maxAge time.Duration, queryFactory *SearchQueryBuilder) ([]string, error) {
+	es.instancesMu.Lock()
+	defer es.instancesMu.Unlock()
+
+	if time.Since(es.lastInstancesQuery) <= maxAge {
```

Comment:
the query from the queryFactory is not taken into account... this might lead to inconsistencies in the projection table
e.g. the first projection has already been projected once but if another projection, which was never initialized with data queries for instanceids in the timeframe it would only get the reduced instanceids and not the complete list
it could also be the other way around, which would not lead to inconsistencies, but rather unnecessary processing of all instances (not much of a problem)

Reply:
should be fixed